lib/rails_admin/support/csv_converter.rb

```diff
@@ -71,13 +71,22 @@ def export_field_for(method, model_config = @model_config)
 
     def generate_csv_string(options)
       generator_options = (options[:generator] || {}).symbolize_keys.delete_if { |_, value| value.blank? }
-      method = @objects.respond_to?(:find_each) ? :find_each : :each
 
       CSV.generate(**generator_options) do |csv|
         csv << generate_csv_header unless options[:skip_header]
 
-        @objects.send(method) do |object|
-          csv << generate_csv_row(object)
+        if @objects.respond_to?(:page)
```

Comment:
```rb
if @objects.respond_to?(:page)
  page_num = 1
  loop do
    batch = @objects.page(page_num)
    break if batch.empty?

    batch.each { |object| csv << generate_csv_row(object) }
    page_num += 1
  end
else
  @objects.each { |object| csv << generate_csv_row(object) }
end
    batch = @objects.page(page_num)
    break if batch.empty?

    batch.each { |object| csv << generate_csv_row(object) }
    page_num += 1
  end
else
  @objects.each { |object| csv << generate_csv_row(object) }
end
```

- Replaced the while loop with a loop and break construct to make the code more readable and maintainable.
- Instead of using batch.any? to check if there are elements in the batch, Used batch.empty? This is more explicit and easier to understand.
- Moved the **page_num += 1** statement to the end of the loop to avoid repetition.